File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {

        std::unordered_map<int, int> numCount;

        for (int i = 0; i < nums.size(); ++i) {
            numCount[nums[i]]++;
            if (numCount[nums[i]] > 1) {
                return true;
            }
        }
        
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqs = defaultdict(int)
        
        orderedList = []
        res = []

        for n in nums:
            freqs[n] += 1
        


        for n in freqs:
            orderedList.append((n, freqs[n]))
            orderedList.sort(key = lambda p: p[1], reverse = True)
        
        for i in range(k):
            res.append(orderedList[i][0])

        return res